### PR TITLE
Document Special Defense badge boost glitch

### PR DESCRIPTION
**Pull request by bbbbbbbbba**
See https://github.com/pret/pokecrystal/issues/643#issuecomment-531966924.

### CODE REVIEW

**Comment by xCrystal** [comment]
While we're at it, the header says:

```
; Every other badge boosts a stat, starting from the first.

; 	ZephyrBadge:  Attack
; 	PlainBadge:   Speed
; 	MineralBadge: Defense
; 	GlacierBadge: Special Attack
; 	RisingBadge:  Special Defense
```

But RisingBadge does not boost Special Defense.

File: docs/bugs_and_glitches.md [hunk 1]
```diff
@@ -167,6 +168,34 @@ This bug existed for all battles in Gold and Silver, and was only fixed for sing
 (This fix also affects Thick Club, Light Ball, and Metal Powder, as described above, but their specific fixes in the above bugs allow more accurate damage calculations.)
 
 
+## Glacier Badge may not boost Special Defense depending on the value of Special Attack
+
+As Pryce's dialogue ("That BADGE will raise the SPECIAL stats of POKéMON.") implies, Glacier Badge is intended to boost both Special Attack and Special Defense. However, due to BoostStat overwriting `a` when boosting Special attack, the Special Defense boost will not happen if the unboosted Special Attack stat is either 0 ~ 205, or 433 ~ 660.
```
Comment (Rangi42) [inline docs/bugs_and_glitches.md]: Nitpick: nowhere else uses "dialogue", always "dialog".

Comment (Rangi42) [inline docs/bugs_and_glitches.md]: "Special attack" -> "Special Attack"

Comment (Rangi42) [inline docs/bugs_and_glitches.md]: "0-205 or 433-660" would match the range notation used elsewhere.

**Comment by Rangi42** [comment]
Good find! I thought we had identified all the bugs, but no.

**Comment by bbbbbbbbba** [comment]
Thanks for checking! All fixed.

**Pull request merged**